packages/x-date-pickers/src/internals/hooks/useField/useField.interfaces.ts

```diff
@@ -290,6 +291,14 @@ export interface UseFieldState<TValue, TSection extends FieldSection> {
    */
   referenceValue: TValue;
   sections: TSection[];
+
+  /**
+   * The separator to insert before the first section.
+   * If it contains escaped characters, then it must not have the escaping characters.
+   * For example, on Day.js, the format `[Current year:] YYYY` has a `startSeparator` equal to `Current year:` not `[Current year:]`
+   */
+  startSeparator: string;
```

Comment:
The alternative would be to put a `startSeparator` / `endSeparator` on each section (to replace the `separator` which is always at the end)

I did not wanted to do that because only the 1st section is impacted.
But with single input range fields, it becomes more complicated since the 1st section of the 2nd date has the separator as well.
So maybe, the tradeoff is not that good...

Reply:
The other option would be to put the `startSeparator` of the second date at the end of the first date.

But having a prefix/suffix for all section even if most of the time they are empty sounds more reliable.

It will also avoid those `getSectionsFromValue(...).sections`